Kernel/Modules/AgentTicketCompose.pm

```diff
@@ -473,7 +474,7 @@ sub Run {
 
     # cycle trough the activated Dynamic Fields for this screen
     DYNAMICFIELD:
-    for my $DynamicFieldConfig ( @{ $Self->{DynamicField} } ) {
+    for my $DynamicFieldConfig ( values $Self->{DynamicField}->%* ) {
         next DYNAMICFIELD if !IsHashRefWithData($DynamicFieldConfig);
 
         # extract the dynamic field value from the web request
```

Comment:
Prio not high, but the lines below `# convert dynamic field values into a structure for ACLs` (490ff - I cannot "review" there, exactly) could be aligned e.g. with ActionCommon.